src/Microsoft.DocAsCode.Build.RestApi/Swagger/Internals/SwaggerJsonBuilder.cs

```diff
@@ -255,5 +289,16 @@ private static SwaggerObject ResolveSwaggerObject(SwaggerObjectBase swaggerObjec
 
             throw new ArgumentException($"When resolving reference for {nameof(SwaggerReferenceObject)}, only support {nameof(SwaggerObject)} and {nameof(SwaggerReferenceObject)} as parameter.");
         }
+
+        private static string GetDirectory(string path)
+        {
+            var directory = Path.GetDirectoryName(path);
+            if (string.IsNullOrEmpty(directory))
+            {
+                throw new DocfxException($"Directory of swagger file path {path} should not be null or empty.");
```

Comment:
is it better to not highly dependent on the caller?

---
In reply to: [101679374](https://github.com/dotnet/docfx/pull/1308#discussion_r101679374) [](ancestors = 101679374,101675921)